docs/proposals/performance.md

```diff
@@ -0,0 +1,21 @@
+# Post-1.0 performance/scalability improvement ideas
+
```

Comment:
all ideas mentioned are master-y, did you mean for this to include ideas about other parts of our system as well?